### PR TITLE
Ensure generic parameter constraint type is included when building image

### PR DESCRIPTION
**Pull request by UnityAlex**
sets.

This fixes: https://issuetracker.unity3d.com/issues/editor-crash-when-entering-and-exiting-playmode-for-a-few-times-with-microsoft-maps-sdk

@joncham and I were working on this we hit the two asserts that I've removed. We were wondering if they were still needed?

### CODE REVIEW

**Comment by vargaz** [comment]
Why did the asserts fail ? Did they fail before the change or after ?

**Comment by UnityAlex** [comment]
> 
> 
> Why did the asserts fail ? Did they fail before the change or after ?

After. It wasn't immediately clear to us how our change caused them to fail. We were hoping someone here would be more familiar with the intent of the asserts and could help us out.

**Comment by vargaz** [comment]
type_in_image () might need similar changes to fix the assertions.

These asserts make sure that when we free a generic instance because an image is unloaded, that the instance actually referenced the image.

**Comment by UnityAlex** [comment]
@vargaz that did the trick. I've updated type_in_image and re-added the asserts.

File: mono/metadata/metadata.c [hunk 1]
```diff
@@ -2570,7 +2570,14 @@ type_in_image (MonoType *type, MonoImage *image)
 		return signature_in_image (type->data.method, image);
 	case MONO_TYPE_VAR:
 	case MONO_TYPE_MVAR:
-		return image == mono_get_image_for_generic_param (type->data.generic_param);
+		if (image == mono_get_image_for_generic_param (type->data.generic_param))
+			return TRUE;
+		else if (type->data.generic_param->gshared_constraint)
+		{
+			type = type->data.generic_param->gshared_constraint;
```
Comment (vargaz) [inline mono/metadata/metadata.c:2577]: K&R Coding convention.

Comment (UnityAlex) [reply mono/metadata/metadata.c:2576]: Updated to use the correct coding style

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**